src/js/login/utils/helpers.js

```diff
@@ -0,0 +1,65 @@
+import Raven from 'raven-js';
+import appendQuery from 'append-query';
+
+import { apiRequest } from '../../common/helpers/api';
+import environment from '../../common/helpers/environment';
+
+const SESSIONS_URI = `${environment.API_URL}/sessions`;
+const redirectUrl = (type) => `${SESSIONS_URI}/${type}/new`;
+
+const MHV_URL = redirectUrl('mhv');
+const DSLOGON_URL = redirectUrl('dslogon');
+const IDME_URL = redirectUrl('idme');
+const MFA_URL = redirectUrl('mfa');
+const VERIFY_URL = redirectUrl('verify');
+const LOGOUT_URL = redirectUrl('slo');
+
+const loginUrl = (policy) => {
+  switch (policy) {
+    case 'mhv': return MHV_URL;
+    case 'dslogon': return DSLOGON_URL;
+    default: return IDME_URL;
+  }
+};
+
+function popup(popupUrl, clickedEvent, openedEvent) {
+  window.dataLayer.push({ event: clickedEvent });
+  const popupWindow = window.open('', 'vets.gov-popup', 'resizable=yes,scrollbars=1,top=50,left=500,width=500,height=750');
+  if (popupWindow) {
+    window.dataLayer.push({ event: openedEvent });
+    popupWindow.focus();
+
+    return apiRequest(
+      popupUrl,
+      null,
+      ({ url }) => { popupWindow.location = url; },
+      () => { popupWindow.location = `${environment.BASE_URL}/auth/login/callback`; }
+    ).then(() => Promise.resolve(popupWindow));
```

Comment:
You should be able to return popupWindow here without the Promise.resolve, I think.